routing/http/types/iter/iter.go

```diff
@@ -0,0 +1,28 @@
+package iter
+
+// Iter is an iterator of aribtrary values.
+// Iterators are generally not goroutine-safe, to make them safe just read from them into a channel.
+// For our use cases, these usually have a single reader. This motivates iterators instead of channels,
+// since the overhead of goroutines+channels has a significant performance cost.
+// Using an iterator, you can read results directly without necessarily involving the Go scheduler.
+type Iter[T any] interface {
+	// Next returns the next element, true if an attempt was made to get the next element, and any error that occurred.
+	Next() (T, bool, error)
```

Comment:
We should add a Close() method to avoid the need of exhausting the iterator to free resources.